ext/session/session.c

```diff
@@ -485,19 +485,23 @@ PHPAPI int php_session_valid_key(const char *key) /* {{{ */
 /* }}} */
 
 
-static void php_session_gc(void) /* {{{ */
+static zend_long php_session_gc(zend_bool immediate) /* {{{ */
 {
 	int nrand;
+	int num = -1;
 
 	/* GC must be done before reading session data. */
 	if ((PS(mod_data) || PS(mod_user_implemented)) && PS(gc_probability) > 0) {
-		int nrdels = -1;
-
+		if (immediate) {
```

Comment:
Doesn't this beat the point of checking `gc_probability`? If so, shouldn't it be outside of the previous `if()` condition?